### PR TITLE
Uplift Enable support of Unstoppable Domains and ENS via DoH on Android to 1.24.x

### PR DESCRIPTION
**Pull request by yrliou**
Uplift of #8520 (Ethereum support is removed from this uplift PR)
Resolves https://github.com/brave/brave-browser/issues/15159

Pre-approval checklist: 
- [x] You have tested your change on Nightly.
- [x] This contains text which needs to be translated. 
    - [ ] There are more than 7 days before the release. 
    - [ ] I've notified folks in #l10n on Slack that translations are needed. 
- [x] The PR milestones match the branch they are landing to. 

Pre-merge checklist: 
- [x] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 

Post-merge checklist: 
- [x] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by yrliou** [comment]
trivial conflict resolving for android/java/org/chromium/chrome/browser/preferences/BravePrefServiceBridge.java, skipping CI expect for android.

**Review by kjozwiak — APPROVED** [review]
Uplift into `1.24.x` approved after deliberating with @brave/uplift-approvers. QA has verified the PR on Nightly as per https://github.com/brave/brave-core/pull/8520#issuecomment-826987500.

**Pull request merged**
